tests/unit/test_infos.py

```diff
@@ -676,3 +690,200 @@ def test_get_host_architecture_returns_valid_arch(
     result = infos._get_host_architecture()
 
     assert result == translated_machine
+
+
+class TestProjectVarInfo:
+    """Tests for the ProjectVarInfo class."""
+
+    def test_marshal_unmarshal(self):
+        """Marshal and unmarshal a ProjectVarInfo."""
+        data = {
+            "a": {
+                "b": {
+                    "value": "value1",
+                },
+                "c": {
+                    "d": {
+                        "value": "value2",
+                        "updated": True,
+                    }
+                },
+            },
+            "e": {
+                "value": "value3",
+                "updated": False,
+            },
+        }
+
+        info = ProjectVarInfo.unmarshal(data)
+        new_data = info.marshal()
+
+        assert new_data == {
+            "a": {
+                "b": {
+                    "value": "value1",
+                    "updated": False,
+                },
+                "c": {
+                    "d": {
+                        "value": "value2",
+                        "updated": True,
+                    },
+                },
+            },
+            "e": {
+                "value": "value3",
+                "updated": False,
+            },
+        }
+
+    def test_unmarshal_error(self):
+        """Error if unmarshalling invalid data."""
+        expected_error = "Project variable info must be a dictionary."
+
+        with pytest.raises(TypeError, match=expected_error):
+            ProjectVarInfo.unmarshal("invalid")  # pyright: ignore[reportArgumentType]
+
+    def test_has_key(self, mock_logger):
+        """Check if a key exists."""
+        info = ProjectVarInfo.unmarshal({"a": {"b": {"value": "value"}}})
+
+        assert info.has_key("a", "b")
```

Comment:
Is the final key also considered part of the structured path? It could be interesting to also test `info.has_key("a", "b", "value")`.